File: app/src/main/java/com/datarockets/mnchkn/fragments/players/di/PlayerComponent.java
package com.datarockets.mnchkn.fragments.players.di;

import com.datarockets.mnchkn.di.ActivityScope;
import com.datarockets.mnchkn.di.AppComponent;
import com.datarockets.mnchkn.fragments.players.PlayerFragment;

import dagger.Component;

@ActivityScope
@Component(dependencies = AppComponent.class, modules = PlayerModule.class)
public interface PlayerComponent {
    void inject(PlayerFragment playerFragment);
}

File: app/src/main/kotlin/com/datarockets/mnchkn/fragments/players/di/PlayerComponent.kt
package com.datarockets.mnchkn.fragments.players.di

import com.datarockets.mnchkn.di.ActivityScope
import com.datarockets.mnchkn.di.AppComponent
import com.datarockets.mnchkn.fragments.players.PlayerFragment

import dagger.Component

@ActivityScope
@Component(dependencies = arrayOf(AppComponent::class), modules = arrayOf(PlayerModule::class))
interface PlayerComponent {
    fun inject(playerFragment: PlayerFragment)
}
